src/WidgetBase.ts

```diff
@@ -421,8 +464,8 @@ export class WidgetBase<P extends WidgetProperties = WidgetProperties, C extends
 		return this._cachedVNode;
 	}
 
-	public invalidate(): void {
-		if (this._renderState === WidgetRenderState.IDLE) {
+	public invalidate(force?: boolean): void {
```

Comment:
I will let @rorticus correct me (or maybe @matt-gadd) but I think is to support the mechanism which forces an immediate re-render when the meta is can only be collected once the Elements are created (and therefore missed by the first render). With the `renderState` introduced, normal invalidation triggered in the `render` lifecycle is a noop and therefore `force` allows that to be overridden in special circumstances.